### ISSUE TITLE
(Question) Usage with QMK/VIA Keyboards

### ISSUE THREAD

**Issue by jayhemsley** [RedBearAK/toshy#414]
I currently use Toshy with a Magic Keyboard, but I'm considering moving to a keyboard that supports QMK/VIA. I've never used that before, but would I still need to use Toshy if I remap the keys via that instead? I'm not 100% sure how they would work together or if it'd interfere with Toshy and if I'm better off just using Toshy alone.

**Comment by RedBearAK** [comment]
@jayhemsley 

If you take a look at the default Toshy config file, you may notice that it is getting close to 5,000 lines long. Ignoring about 1,500 lines for the "OptSpec" special characters scheme that almost nobody uses, there are still tons of remaps for various types of Linux applications. Specific apps and app groups like "browsers" and "file managers". 

In other words, the config does a lot more than moving modifiers around. Moving the modifiers takes care of a few general shortcuts, but there are also a few dozen combo remaps for other general shortcuts, plus quite a few that are specific to different desktop environments, and terminals. 

With a custom keyboard I'd recommend the same as a universal keyboard that can be switched between Mac and Windows modes. Leave it in a "Windows" modifier configuration, and it should work perfectly with Toshy. The Toshy config only has modmaps for Apple, Windows, IBM and Chromebook keyboard types. Besides official Apple/Magic keyboards, most keyboards default to the "Windows" type unless you manually override that in the config file. There's a FAQ entry about how to fix the keyboard type, but if you use some oddball set of modifier positions you'd need to create your own modmap. 

On the other hand, it would probably be easier to do things like making CapsLock act as Escape and Command using the programmability of the keyboard software. So the software/firmware of the keyboard could still do handy things independently of what Toshy does. You can do that in the config file, but the multipurpose modmaps seem a little wonky in practice. 

In short, you would be missing out on a lot if you just moved the modifiers and didn't use the whole Toshy config. 

I would take a look in particular at the "General" keymap at the end of the config file. If you use any of those remaps or the adjunct remaps in DE-specific keymaps just above that, those would be gone if you weren't using Toshy. 

Let me know if this information was helpful.

**Comment by jayhemsley** [comment]
@RedBearAK Okay thanks! I'll check it out. The keyboards I'm looking at use the Magic Keyboard layout more or less anyway, so I figure it should be a pretty easy transition.
